fix OMParser/Makefile for macOS

alexchandel opened the pull request:
Remove breaking linker flags for macOS. macOS has no `static-libgcc`, `-lstdc++` is unnecessary for macOS's clang, and `-lantlr4-runtime` this early causes CMake's build system check to fail.

alexchandel commented:
This is necessary to unbreak the macOS build.  Otherwise clang errors out with `unsupported option '-static-libgcc'`.

adrpo commented:
Hm, these flags are not needed at all in the OMParser/Makefile only in OMParser/test/Makefile. 
Just remove them,  both CXXFLAGS and LDFLAGS. I think I just left them there by mistake. 
Strangely, our Mac mini doesn't complain, maybe the cmake we have ignores these flags?!

Please remove the flags and do a new commit. I added the `CI/Build OSX` label so we can test it on the mac as well.

sjoelund commented:
> Strangely, our Mac mini doesn't complain, maybe the cmake we have ignores these flags?!

Or our Mac Mini is older and still has the GCC legacy stuff... Apple overhauled everything several times already.

adrpo commented:
@sjoelund then I guess is time to upgrade everything on it, is that even possible or we need a new one?

sjoelund commented:
We can upgrade the OS once we're allowed at LiU. mac00286 is more recent and faster so you could just change the target to that machine

alexchandel commented:
I removed the LDFLAGS. I find I still need the CXXFLAGS to build OMParser itself (tho not ANTLR).

Also, I added a `mkdir -p`, as building `omparser` before `omc` fails with something like:
```
cp -pR libomcparserantlr4.a /private/tmp/openmodelica-20210302-26076-13f29jp/build/lib/x86_64-darwin20.3.0/omc/
cp: directory /private/tmp/openmodelica-20210302-26076-13f29jp/build/lib/x86_64-darwin20.3.0/omc does not exist
```

Possible CMake is finding GCC somewhere in your path.  Also possible you already have a `libantlr-runtime` somewhere in your paths, which it finds.  I've been using a Homebrew formula (72254) to emulate vanilla build conditions to find some of this stuff.
